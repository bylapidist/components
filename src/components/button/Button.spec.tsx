import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import {
    Button,
    DangerButton,
    PrimaryButton,
    SecondaryButton,
    TertiaryButton
} from './index';

test('it works as default', () => {
    const tree = renderer.create(<Button text="Click me" />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as primary', () => {
    const tree = renderer.create(<PrimaryButton text="Click me" />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as secondary', () => {
    const tree = renderer.create(<SecondaryButton text="Click me" />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as tertiary', () => {
    const tree = renderer.create(<TertiaryButton text="Click me" />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as danger', () => {
    const tree = renderer.create(<DangerButton text="Click me" />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as block', () => {
    const tree = renderer.create(<Button text="Click me" block />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as block loading', () => {
    const tree = renderer
        .create(<Button text="Click me" block loading />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as block outline', () => {
    const tree = renderer
        .create(<Button text="Click me" block outline />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as disabled', () => {
    const tree = renderer.create(<Button text="Click me" disabled />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as outline small', () => {
    const tree = renderer
        .create(<Button text="Click me" outline small />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as small loading', () => {
    const tree = renderer
        .create(<Button text="Click me" small loading />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('it works as outline small disabled', () => {
    const tree = renderer
        .create(<Button text="Click me" outline disabled small />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
