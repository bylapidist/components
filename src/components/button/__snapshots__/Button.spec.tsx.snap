// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #66687d;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="#66687d"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as block 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #66687d;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="#66687d"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as block loading 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: transparent;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={true}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="transparent"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as block outline 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: transparent;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="transparent"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as danger 1`] = `
.c1 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #b41e25;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 {
  position: relative;
  -webkit-transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  font-family: 'Montserrat',sans-serif;
  font-size: 16px;
  outline: 2px transparent;
  outline-offset: 2px;
  background: none;
  border: none;
  height: 57px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  margin-right: 14px;
  margin-bottom: 14px;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  color: #ffffff;
}

.c0:focus {
  outline: 2px solid #ebc764;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.8;
  color: #b41e25;
}

.c0:hover:not(:disabled) {
  cursor: pointer;
}

<button
  className="c0"
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c1"
    color="#b41e25"
  >
    <span
      className="c2"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as default 1`] = `
.c1 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #66687d;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 {
  position: relative;
  -webkit-transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  font-family: 'Montserrat',sans-serif;
  font-size: 16px;
  outline: 2px transparent;
  outline-offset: 2px;
  background: none;
  border: none;
  height: 57px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  margin-right: 14px;
  margin-bottom: 14px;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  color: #ffffff;
}

.c0:focus {
  outline: 2px solid #ebc764;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.8;
  color: #66687d;
}

.c0:hover:not(:disabled) {
  cursor: pointer;
}

<button
  className="c0"
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c1"
    color="#66687d"
  >
    <span
      className="c2"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as disabled 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #66687d;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={true}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="#66687d"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as outline small 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: transparent;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="transparent"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as outline small disabled 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: transparent;
  border: 2px #66687d;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={true}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="#66687d"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as primary 1`] = `
.c1 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #17a3a5;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 {
  position: relative;
  -webkit-transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  font-family: 'Montserrat',sans-serif;
  font-size: 16px;
  outline: 2px transparent;
  outline-offset: 2px;
  background: none;
  border: none;
  height: 57px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  margin-right: 14px;
  margin-bottom: 14px;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  color: #ffffff;
}

.c0:focus {
  outline: 2px solid #ebc764;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.8;
  color: #17a3a5;
}

.c0:hover:not(:disabled) {
  cursor: pointer;
}

<button
  className="c0"
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c1"
    color="#17a3a5"
  >
    <span
      className="c2"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as secondary 1`] = `
.c1 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #72af5d;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 {
  position: relative;
  -webkit-transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  font-family: 'Montserrat',sans-serif;
  font-size: 16px;
  outline: 2px transparent;
  outline-offset: 2px;
  background: none;
  border: none;
  height: 57px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  margin-right: 14px;
  margin-bottom: 14px;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  color: #ffffff;
}

.c0:focus {
  outline: 2px solid #ebc764;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.8;
  color: #72af5d;
}

.c0:hover:not(:disabled) {
  cursor: pointer;
}

<button
  className="c0"
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c1"
    color="#72af5d"
  >
    <span
      className="c2"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as small loading 1`] = `
.c0 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: transparent;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<button
  disabled={true}
  type="button"
>
  <div
    className="sc-AxjAm c0"
    color="transparent"
  >
    <span
      className="c1"
    >
      Click me
    </span>
  </div>
</button>
`;

exports[`it works as tertiary 1`] = `
.c1 {
  box-shadow: 0 2px 10px rgba(243,245,249,0.025),0 2px 10px rgba(243,245,249,0.035);
  border-radius: 0;
  background: #ebc764;
  border: 2px transparent;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  text-align: center;
  line-height: 1.18;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  position: absolute;
  top: 50%;
  left: 0;
  right: 0;
  padding: 0 14px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c0 {
  position: relative;
  -webkit-transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  transition: all 0.12s cubic-bezier(0.6,0.04,0.98,0.7);
  font-family: 'Montserrat',sans-serif;
  font-size: 16px;
  outline: 2px transparent;
  outline-offset: 2px;
  background: none;
  border: none;
  height: 57px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: auto;
  margin-right: 14px;
  margin-bottom: 14px;
  padding: 0;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 1;
  color: #ffffff;
}

.c0:focus {
  outline: 2px solid #ebc764;
}

.c0:disabled {
  cursor: not-allowed;
  opacity: 0.8;
  color: #ebc764;
}

.c0:hover:not(:disabled) {
  cursor: pointer;
}

<button
  className="c0"
  disabled={false}
  type="button"
>
  <div
    className="sc-AxjAm c1"
    color="#ebc764"
  >
    <span
      className="c2"
    >
      Click me
    </span>
  </div>
</button>
`;
